core/lib/eth_client/src/types.rs

```diff
@@ -129,6 +134,64 @@ pub struct SignedCallResult {
     pub hash: H256,
 }
 
+impl SignedCallResult {
+    pub fn new(
+        raw_tx: RawTransactionBytes,
+        max_priority_fee_per_gas: U256,
+        max_fee_per_gas: U256,
+        nonce: U256,
+        hash: H256,
+    ) -> Self {
+        Self {
+            raw_tx,
+            max_priority_fee_per_gas,
+            max_fee_per_gas,
+            nonce,
+            hash,
+        }
+    }
+
+    pub fn raw_tx(&self, blob_tx_sidecar: Option<&EthTxBlobSidecar>) -> RawTransactionBytes {
+        match blob_tx_sidecar {
+            None => self.raw_tx.clone(),
+            Some(sidecar) => encode_blob_tx_with_sidecar(&self.raw_tx, sidecar),
+        }
+    }
+}
+
+fn encode_blob_tx_with_sidecar(
+    raw_tx: &RawTransactionBytes,
+    sidecar: &EthTxBlobSidecar,
+) -> RawTransactionBytes {
+    let EthTxBlobSidecar::EthTxBlobSidecarV1(sidecar) = sidecar;
+    let blobs_count = sidecar.blobs.len();
+
+    let raw_tx = &raw_tx.0;
+    let mut stream_outer = RlpStream::new();
+    stream_outer.begin_list(4);
```

Comment:
comment where 4 comes from (and preferably replace with a constant)

Reply:
comment is added